BibBuddy/app/src/main/java/de/bibbuddy/TextNoteEditorFragment.java

```diff
@@ -22,6 +30,43 @@
   private Note note;
   private NoteModel noteModel;
 
+  @Override
+  public void onCreate(Bundle savedInstanceState)
+  {
+    super.onCreate (savedInstanceState);
+    requireActivity().getOnBackPressedDispatcher().addCallback(new OnBackPressedCallback(true) {
+      @Override
+      public void handleOnBackPressed() {
+        saveNote();
+        FragmentManager fm = getParentFragmentManager();
+        if (fm.getBackStackEntryCount() > 0) {
+          fm.popBackStack();
+        } else {
+          requireActivity().onBackPressed();
+        }
+      }
+    });
+  }
+
+  private void saveNote() {
+    String text = Html.toHtml(richTextEditor.getText(), Html.TO_HTML_PARAGRAPH_LINES_INDIVIDUAL);
+    String rawText = Jsoup.parse(text).text();
+    String name = "";
+    BufferedReader bufferedReader = new BufferedReader(new StringReader(rawText));
+    try {
+      name = bufferedReader.readLine();
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    if (rawText.length() != 0) {
+      if(getArguments() != null) {
+        noteModel.updateNote(note, name, text);
+      } else {
+        noteModel.addNote(name, 0 , text);
+      }
+    }
+  }
```

Comment:
BookId should also be saved, which you get via a bundle.

Reply:
This needs multiple adjustments to other classes (especially bookView related ones) - but it is done now.